blosc/CMakeLists.txt

```diff
@@ -88,7 +88,32 @@ endif(WIN32)
 
 if(NOT DEACTIVATE_LZ4)
     if(LZ4_FOUND)
-        set(LIBS ${LIBS} ${LZ4_LIBRARY})
+        # Check if LZ4 was detected from a conig
+        # module first
+        if(TARGET LZ4::lz4_shared)
+            set(SHARED_LIBS LZ4::lz4_shared)
+        endif()
+        if(TARGET LZ4::lz4_static)
+            set(STATIC_LIBS LZ4::lz4_static)
+        endif()
+        if(SHARED_LIBS AND STATIC_LIBS)
+            # Set a genex to match the appropriate library type respective to the
+            # type of the LZ4 library
+            set(LIBS
+                ${LIBS}
+                $<$<STREQUAL:$<TARGET_PROPERTY:Foo,TYPE>,SHARED_LIBRARY>:LZ4::lz4_shared>
+                $<$<NOT:$<STREQUAL:$<TARGET_PROPERTY:Foo,TYPE>,SHARED_LIBRARY>>:LZ4::lz4_static>
+            )
+        elseif(SHARED_LIBS OR STATIC_LIBS)
+            if(SHARED_LIBS)
+                set(LIBS ${LIBS} ${SHARED_LIBS})
+            else()
+                set(LIBS ${LIBS} ${STATIC_LIBS})
+            endif()
+        elseif(NOT SHARED_LIBS AND NOT STATIC_LIBS)
+            # Fallback to cblosc vendored find module
```

Comment:
Isn't the vendored case handled below, after `else(LZ4_FOUND)`?

Reply:
That's the case where you vendor `LZ4` in its entirety, this handles the case where the `c-blosc` vendored find module for `LZ4` is used instead of `LZ4`'s config module.

Reply:
Different components being vendored, I had hoped the "vendored find module" would be clear enough, but evidently it was not, my apologies.

Reply:
I'm not familiar with libraries providing cmake snippets, i only heard about pkg-config. I'll let someone more familiar with Cmake review. To me, the "genex" code is unreadable and gives vibes of a the xz backdoor...

Reply:
CMake's Config modules are an integral part of a healthy CMake ecosystem, it allows you to export your project in a way that allows consumers to use your project exactly as you intend. 

[The doc](https://cmake.org/cmake/help/latest/guide/importing-exporting/index.html)

Reply:
I did just notice a mistake in the generator expressions, but that doesn't mean they're a vulnerability...

Reply:
Which is now fixed, forgot to refactor the code from my test project.